app/bundles/LeadBundle/EventListener/FormSubscriber.php

```diff
@@ -39,45 +58,71 @@ public function onFormBuilder(FormBuilderEvent $event)
         ];
         $event->addSubmitAction('lead.pointschange', $action);
 
         //add to lead list
         $action = [
-            'group'       => 'mautic.lead.lead.submitaction',
-            'label'       => 'mautic.lead.lead.events.changelist',
+            'group' => 'mautic.lead.lead.submitaction',
+            'label' => 'mautic.lead.lead.events.changelist',
             'description' => 'mautic.lead.lead.events.changelist_descr',
-            'formType'    => 'leadlist_action',
-            'callback'    => '\Mautic\LeadBundle\Helper\FormEventHelper::changeLists',
+            'formType' => 'leadlist_action',
+            'callback' => '\Mautic\LeadBundle\Helper\FormEventHelper::changeLists',
         ];
         $event->addSubmitAction('lead.changelist', $action);
 
         // modify tags
         $action = [
-            'group'             => 'mautic.lead.lead.submitaction',
-            'label'             => 'mautic.lead.lead.events.changetags',
-            'description'       => 'mautic.lead.lead.events.changetags_descr',
-            'formType'          => 'modify_lead_tags',
-            'callback'          => '\Mautic\LeadBundle\Helper\EventHelper::updateTags',
+            'group' => 'mautic.lead.lead.submitaction',
+            'label' => 'mautic.lead.lead.events.changetags',
+            'description' => 'mautic.lead.lead.events.changetags_descr',
+            'formType' => 'modify_lead_tags',
+            'callback' => '\Mautic\LeadBundle\Helper\EventHelper::updateTags',
             'allowCampaignForm' => true,
         ];
         $event->addSubmitAction('lead.changetags', $action);
 
         // add UTM tags
         $action = [
-            'group'       => 'mautic.lead.lead.submitaction',
-            'label'       => 'mautic.lead.lead.events.addutmtags',
+            'group' => 'mautic.lead.lead.submitaction',
+            'label' => 'mautic.lead.lead.events.addutmtags',
             'description' => 'mautic.lead.lead.events.addutmtags_descr',
-            'formType'    => 'lead_action_addutmtags',
-            'formTheme'   => 'MauticLeadBundle:FormTheme\\ActionAddUtmTags',
-            'callback'    => '\Mautic\LeadBundle\Helper\EventHelper::addUtmTags',
+            'formType' => 'lead_action_addutmtags',
+            'formTheme' => 'MauticLeadBundle:FormTheme\\ActionAddUtmTags',
+            'callback' => '\Mautic\LeadBundle\Helper\EventHelper::addUtmTags',
         ];
         $event->addSubmitAction('lead.addutmtags', $action);
+
+        // add Do Not Contact
+        $action = [
+            'group' => 'mautic.lead.lead.submitaction',
+            'label' => 'mautic.lead.lead.events.removedonotcontact',
+            'description' => 'mautic.lead.lead.events.removedonotcontact_descr',
+            'formType' => 'lead_action_removedonotcontact',
+            'formTheme' => 'MauticLeadBundle:FormTheme\\ActionRemoveDoNotContact',
+            'eventName' => FormEvents::FORM_ON_SUBMIT,
```

Comment:
@alanhartless Should this be a unique event name instead of `FORM_ON_SUBMIT`? We don't want it to run all the events registered to `FormEvents::FORM_ON_SUBMIT` when the lead.removeDoNotContact action is triggered, we just want it to trigger the onFormSubmit method below.

From reading, it looks like every time a form is submitted it will remove DNC records for the contact.